src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -2480,7 +2522,19 @@ module Server =
               let req = JsonRpc.Request.Create(getNextRequestId(), rpcMethod, serializedRequest)
               let reqString = JsonConvert.SerializeObject(req, jsonSettings)
               sender.Post(reqString)
-              let! response = responseAgent.PostAndAsyncReply((fun replyChannel -> Request(req.Id, replyChannel)))
+              use cts = new System.Threading.CancellationTokenSource()
```

Comment:
I'd like to encapsulate this pattern somehow, where we create a scoped cts and perform some request with that cts's token registered.